action.yml

```diff
@@ -1,17 +1,21 @@
 name: 'Setup Node.js environment'
-description: 'Setup a Node.js environment and add it to the PATH, additionally providing proxy support'
+description: 'Setup a Node.js environment by adding problem matchers and optionally downloading and adding it to the PATH'
 author: 'GitHub'
 inputs:
   always-auth:
     description: 'Set always-auth in npmrc'
     default: 'false'
   node-version:
-    description: 'Version Spec of the version to use.  Examples: 10.x, 10.15.1, >=10.15.0'
-    default: '12.x'
+    description: 'Version Spec of the version to use.  Examples: 12.x, 10.15.1, >=10.15.0'
   registry-url:
     description: 'Optional registry to set up for auth. Will set the registry in a project level .npmrc and .yarnrc file, and set up auth to read in from env.NODE_AUTH_TOKEN'
   scope:
     description: 'Optional scope for authenticating against scoped registries'
+  token:
+    description: Used to pull node distributions from node-versions.  Since there's a default, this is typically not supplied by the user.
+    default: ${{ github.token }}
```

Comment:
We need to make sure that querying api.github.com for the manifest and release data doesn't get throttled.  Authenticated is per user and much higher (similar to our runner pulling actions) while anon would be throttled much lower and per IP